nav2_soloco_controller:
  ros__parameters:
    controller_frequency: 20.0 # [Hz]
    # Neural model
    model_name: 'CEM_IAR'
    use_robot_model: True   # Flag for robot param constrains usage
    AR_checkpoint: 'models/weights/SIMNoGoal-univ_fast_AR2/checkpoint_with_model.pt'
    IAR_checkpoint: 'models/weights/SIMNoGoal-univ_IAR_Full_trans/checkpoint_with_model.pt'
    device: 'cpu'           # Options: 'cuda' or 'cpu'
    # Goal tolerance
    goal_tolerance: 0.1
    use_velocity_scaled_lookahead_dist: True # max_lookahead_dist scaled by speed / max_speed
    min_lookahead_dist: 0.4 # Minimum distance to subgoal pose if robot is at very low speed
    max_lookahead_dist: 2.4 # At least max_speed * interp_interval * prediction_steps, must be less than costmap size / 2
    republish_global_goal: False
    republish_goal_period: 0.5
    # Topic parameters
    odom_topic: 'locobot/odom'
    frame_id: 'locobot/odom'
    global_goal_topic: 'goal_pose' # 'global_goal_republished' #   
    subgoal_topic: 'subgoal_pose'
    costmap_topic: 'local_costmap/costmap'
    cmd_vel_topic: 'locobot/commands/velocity'
    robot_track_topic: 'robot/ego_trajectory'
    human_track_topic: 'human/interpolated_history'
    subgoal_marker_topic: 'visualization/subgoal'
    future_topic: 'visualization/predicted_future'
    pub_frame_id: 'locobot/odom'
    # Robot parameters
    robot_model: "differential_drive"
    max_speed: 0.5          # [m/s] # 0.7 locobot kobuki base
    min_speed: -0.1         # [m/s]
    max_yaw_rate: 1.0       # [rad/s]
    max_delta_yaw_rate: 3.2 # [rad/s^2]
    max_accel: 0.7          # [m/s^2] 0.8 kobuki base
    collision_dist: 0.5     # [m]
    max_num_agents: 5       # [maximum number of people]
    max_history_length: 7   # [maximum number of human past poses]
    prediction_steps: 7     # Steps to predict into the future
    interp_interval: 0.4    # [s]
    sample_batch: 100

    debug_log: false